pwnlib/util/packing.py

```diff
@@ -790,42 +791,42 @@ def dd(dst, src, count = 0, skip = 0, seek = 0, truncate = False):
         else:
             src = src.encode('utf8')[seek:]
 
-    elif isinstance(src, file):
+    elif hasattr(src, 'seek'):
```

Comment:
Is there a way that this could be done with `six`?

Reply:
`six` doesn't have a way of doing this. The solution I chose doesn't look ugly to me, because there are several file-like objects even in the standard library that don't subclass `file`, like `ZipFile`, `TarFile` etc. (representing files from inside the archive).

Reply:
I don't think we'd want `ZipFile` or `TarFile` to be used in this case -- they wouldn't really make sense.

We explicitly open a `file` object at the top.  `src` and `dst` can only be one a few specific types -- `file`, `list`, `bytearray`, etc.  If somebody passes a `ZIpFile` they've made an error.

I'd rather replace the very top of this function with something that uses e.g. `six.BytesIO` instead of calling `open()` or `file()`.